controllers/provider-aws/pkg/controlplane/internal/helper_test.go

```diff
@@ -0,0 +1,227 @@
+KubernetesClusterTag="` + namespace + `"
+KubernetesClusterID="` + namespace + `"
+Zone="eu-west-1a"
+`,
+				}
+			)
+
+			// Create mock decoder
+			decoder := mockruntime.NewMockDecoder(ctrl)
+			decoder.EXPECT().Decode(cp.Spec.ProviderConfig.Raw, nil, &apisaws.ControlPlaneConfig{}).DoAndReturn(decode)
+			decoder.EXPECT().Decode(cp.Spec.InfrastructureProviderStatus.Raw, nil, &apisaws.InfrastructureStatus{}).DoAndReturn(decode)
+
+			// Create helper
+			h := NewHelper()
+
+			// Call ConfigChartValuesFunc and check the results
+			values, err := h.ConfigChartValuesFunc(decoder, cp, cluster)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(values).To(Equal(expectedValues))
+		})
+	})
+
+	Describe("#ControlPlaneChartValuesFunc", func() {
+		It("should return correct control plane chart values", func() {
+			var (
+				checksums = map[string]string{
+					cloudControllerManagerDeploymentName: "1",
+					cloudControllerManagerServerName:     "2",
+					common.CloudProviderSecretName:       "3",
+					common.CloudProviderConfigName:       "4",
+				}
+
+				expectedValues = map[string]interface{}{
+					"cloudProvider":     "aws",
+					"clusterName":       namespace,
+					"kubernetesVersion": "1.13.4",
+					"podNetwork":        cidr,
+					"replicas":          1,
+					"podAnnotations": map[string]interface{}{
+						"checksum/secret-cloud-controller-manager":        "1",
+						"checksum/secret-cloud-controller-manager-server": "2",
+						"checksum/secret-cloudprovider":                   "3",
+						"checksum/configmap-cloud-provider-config":        "4",
+					},
+					"configureRoutes": false,
+					"environment": []map[string]interface{}{
+						{
+							"name": "AWS_ACCESS_KEY_ID",
+							"valueFrom": map[string]interface{}{
+								"secretKeyRef": map[string]interface{}{
+									"key":  aws.AccessKeyID,
+									"name": common.CloudProviderSecretName,
+								},
+							},
+						},
+						{
+							"name": "AWS_SECRET_ACCESS_KEY",
+							"valueFrom": map[string]interface{}{
+								"secretKeyRef": map[string]interface{}{
+									"key":  aws.SecretAccessKey,
+									"name": common.CloudProviderSecretName,
+								},
+							},
+						},
+					},
+					"resources": map[string]interface{}{
+						"limits": map[string]interface{}{
+							"cpu":    "500m",
+							"memory": "512Mi",
+						},
+					},
+					"featureGates": map[string]bool{
+						"CustomResourceValidation": true,
+					},
+				}
+			)
+
+			// Create mock decoder
+			decoder := mockruntime.NewMockDecoder(ctrl)
+			decoder.EXPECT().Decode(cp.Spec.ProviderConfig.Raw, nil, &apisaws.ControlPlaneConfig{}).DoAndReturn(decode)
+
+			// Create helper
+			h := NewHelper()
+
+			// Call ControlPlaneChartValuesFunc and check the results
+			values, err := h.ControlPlaneChartValuesFunc(decoder, cp, cluster, checksums)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(values).To(Equal(expectedValues))
+		})
+	})
+
+})
+
+func encode(obj runtime.Object) []byte {
```

Comment:
Error case missing - I know this is test code, but test code should be treated as production code as well. How about directly calling `Expect` in this function and calling it `mustEncode`.